library/src/main/java/com/otaliastudios/zoom/ZoomLogger.kt

```diff
@@ -0,0 +1,99 @@
+package com.otaliastudios.zoom
+
+import android.support.annotation.IntDef
+import android.util.Log
+
+/**
+ * Utility class that can log traces and info.
+ */
+class ZoomLogger private constructor(private val mTag: String) {
+
+    @IntDef(LEVEL_VERBOSE, LEVEL_WARNING, LEVEL_ERROR)
+    @Retention(AnnotationRetention.SOURCE)
+    internal annotation class LogLevel
+
+    internal fun v(message: String) {
+        if (should(LEVEL_VERBOSE)) {
+            Log.v(mTag, message)
+            lastMessage = message
+            lastTag = mTag
+        }
+    }
+
+    internal fun i(message: String) {
+        if (should(LEVEL_INFO)) {
+            Log.i(mTag, message)
+            lastMessage = message
+            lastTag = mTag
+        }
+    }
+
+    internal fun w(message: String) {
+        if (should(LEVEL_WARNING)) {
+            Log.w(mTag, message)
+            lastMessage = message
+            lastTag = mTag
+        }
+    }
+
+    internal fun e(message: String) {
+        if (should(LEVEL_ERROR)) {
+            Log.e(mTag, message)
+            lastMessage = message
+            lastTag = mTag
+        }
+    }
+
+    private fun should(messageLevel: Int): Boolean {
+        return level <= messageLevel
+    }
+
+    private fun string(messageLevel: Int, vararg ofData: Any): String {
+        var message = ""
+        if (should(messageLevel)) {
+            for (o in ofData) {
+                message += o.toString()
+                message += " "
+            }
+        }
+        return message.trim { it <= ' ' }
```

Comment:
I don't know what this does but it creates an object for each Log call right? Let's use simple trim or, better, let's not trim at all..

Reply:
This is generated code from the java -> kotlin conversion. I removed the trim.